File: baseadapter/src/main/java/com/sum/simpleadapter/multiple/Entrust.java
package com.sum.simpleadapter.multiple;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import com.sum.simpleadapter.base.ViewHolder;

import androidx.viewbinding.ViewBinding;

/**
 * @author liujiang
 * Desc:
 */
public interface Entrust<V extends ViewBinding, T> {
    ViewBinding getViewBinding(LayoutInflater layoutInflater, ViewGroup parent);

    boolean isThisType(T t);

    void onBind(Context context, ViewHolder<V> holder, T item, int position);
} 

File: sampleAdapter/src/main/java/com/sum/sampleadapter/multiple/Entrust.kt
package com.sum.sampleadapter.multiple

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.viewbinding.ViewBinding

/**
 * @author  LiuJiang
 * Desc:
 */
abstract class Entrust<V: ViewBinding, T> {

    abstract fun getViewBinding(layoutInflater: LayoutInflater, parent: ViewGroup): V

    abstract fun isThisType(t: T): Boolean

    abstract fun onBind(context: Context, binding: V, position: Int, item: T)
}